framework/sandstone_opts.cpp

```diff
@@ -933,7 +864,436 @@ int parse_cmdline(int argc, char** argv, SandstoneApplication* app, ParsedCmdLin
+            switch (std::get<int>(opts_map[_duration_option])) {
+            case 'T': {
+                auto endtime = std::get<const char*>(opts_map.at('T'));
+                if (strcmp(endtime, "forever") == 0) {
+                    app->endtime = MonotonicTimePoint::max();
+                } else {
+                    app->endtime = app->starttime + string_to_millisecs(endtime);
+                }
+                opts.test_set_config.cycle_through = true; /* Time controls when the execution stops as
+                                                            opposed to the number of tests. */
+                break;
+            }
+            case one_sec_option:
+                opts.test_set_config.randomize = true;
+                opts.test_set_config.cycle_through = true;
+                app->shmem->use_strict_runtime = true;
+                app->endtime = app->starttime + 1s;
+                break;
+            case thirty_sec_option:
+                opts.test_set_config.randomize = true;
+                opts.test_set_config.cycle_through = true;
+                app->shmem->use_strict_runtime = true;
+                app->endtime = app->starttime + 30s;
+                break;
+            case two_min_option:
+                opts.test_set_config.randomize = true;
+                opts.test_set_config.cycle_through = true;
+                app->shmem->use_strict_runtime = true;
+                app->endtime = app->starttime + 2min;
+                break;
+            case five_min_option:
+                opts.test_set_config.randomize = true;
+                opts.test_set_config.cycle_through = true;
+                app->shmem->use_strict_runtime = true;
+                app->endtime = app->starttime + 5min;
+                break;
+            case service_option:
+                // keep in sync with RestrictedCommandLine below
+                opts.fatal_errors = true;
+                app->endtime = MonotonicTimePoint::max();
+                app->service_background_scan = true;
+                break;
+            }
+        }
+
+        // boolean flags
+        opts.fatal_errors = opts_map.count('F');
+
+        opts.test_set_config.ignore_unknown_tests = opts_map.count(ignore_unknown_tests_option);
+        opts.test_set_config.randomize = opts_map.count(test_list_randomize_option);
+
+        app->force_test_time = opts_map.count(force_test_time_option);
+        app->fatal_skips = opts_map.count(fatal_skips_option);
+        app->ignore_mce_errors = opts_map.count(ignore_mce_errors_option);
+        app->ignore_os_errors = opts_map.count(ignore_os_errors_option);
+        app->vary_frequency_mode = opts_map.count(vary_frequency);
+        app->vary_uncore_frequency_mode = opts_map.count(vary_uncore_frequency);
+
+        app->shmem->use_strict_runtime = opts_map.count(strict_runtime_option);
+        app->shmem->ud_on_failure = opts_map.count(ud_on_failure_option);
```

Comment:
These are fine, provided they're not producing warnings on the cast from `size_t` to bool. If they do, just add `!= 0`.